x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/actions.tsx

```diff
@@ -41,7 +43,11 @@ export const duplicateRulesAction = async (
 ) => {
   try {
     dispatch({ type: 'loadingRuleIds', ids: ruleIds, actionType: 'duplicate' });
-    const response = await duplicateRules({ rules });
+    const response = await duplicateRules({
+      // We cast this back and forth here as the front end types are not really the right io-ts ones
+      // and the two types conflict with each other.
+      rules: rules.map((rule) => transformOutput(rule as CreateRulesSchema) as Rule),
```

Comment:
I ran into the same thing with exceptions. I ended up creating two functions to deal with the two type scenarios, but thought about doing the `as` as well.

```js
export const transformOutput = (
  exceptionItem: UpdateExceptionListItemSchema | ExceptionListItemSchema
): UpdateExceptionListItemSchema | ExceptionListItemSchema =>
  flow(removeIdFromExceptionItemsEntries)(exceptionItem);

export const transformNewItemOutput = (
  exceptionItem: CreateExceptionListItemSchema
): CreateExceptionListItemSchema => flow(removeIdFromExceptionItemsEntries)(exceptionItem);
```

Reply:
Thanks for letting me know